services/typeddata/types.go

```diff
@@ -0,0 +1,85 @@
+package typeddata
+
+import (
+	"errors"
+	"fmt"
+	"math/big"
+)
+
+const (
+	eip712Domain = "EIP712Domain"
+	chainIDKey   = "chainId"
+)
+
+// Types define fields for each composite type.
+type Types map[string][]Field
+
+// Field stores name and solidity type of the field.
+type Field struct {
+	Name string `json:"name"`
+	Type string `json:"type"`
+}
+
+// Validate checks that both name and type are not empty.
+func (f Field) Validate() error {
```

Comment:
I didn't find unit test for this method.

Reply:
looks like i missed them will add